ts/nni_manager/training_service/kubernetes/frameworkcontroller/frameworkcontrollerTrainingService.ts

```diff
@@ -59,15 +69,61 @@ class FrameworkControllerTrainingService extends KubernetesTrainingService imple
             }
         }
     }
+    private parseCustomTaskRoles(customTaskRoles: any[]): FrameworkControllerTrialConfigTemplate[] {
+        const taskRoles: FrameworkControllerTrialConfigTemplate[] = []
+        customTaskRoles.map((x) => {
+            if (x.task === undefined ||
+                x.task.pod === undefined ||
+                x.task.pod.spec === undefined ||
+                x.task.pod.spec.containers === undefined) {
+                throw new Error('invalid custom frameworkcontroller configuration')
+            }
+            if (x.task.pod.spec.containers.length > 1) {
+                throw new Error('custom config may only define one non-init container for tasks')
+            }
+            const defaultAttempt = {
+                minFailedTaskCount: 1,
+                minSucceededTaskCount: -1
+            }
+            const trialConfig = <FrameworkControllerTrialConfigTemplate>{
+                name: x.name,
```

Comment:
will x.name be undefined?

Reply:
Using the new validation mechanism, it is ensured, that the name field will never be empty. Is this sufficient or should 
I handle the undefined case at that point anyways?

Reply:
If it is validated in other place, it's ok here.